src/scripts/components/VariableMessage.jsx

```diff
@@ -18,24 +20,43 @@ import {
 } from '../constants';
 
 class VariableMessage extends React.Component {
+  currentDraggedSpanVariable = '';
+  isCategoryMobileViewVisible = false;
   state = {
     message: '',
     available: [],
+    variables: [],
+    variablesOfCategory: [],
+    categories: [],
+    selectedCategory: '',
   };
 
   componentWillMount() {
```

Comment:
i think this would be change with getDerivedStateFromProps(), componentDidMount() can't be use because handleInitValue() method need updated state. May be we also call `handleInitValue` it in this.setState callback, instead of getDerivedStateFromProps(). let me know what you think.

Reply:
@juanfabrega As this is not the part of draggable thing we can make new ticket to handle this -
Remove old lifecycles `componentWillMount, componentWillReceiveProps` with new lifecycles `getDerivedStateFromProps()`.